SparkPost/2web2ui
# FE-922 - Adds new styles for Desktop Navigation

[FE-922](https://jira.int.messagesystems.com/browse/FE-922)

### What Changed
- Implements new navigation design for Hibana and adds some supporting tests:

![Screen Shot 2020-03-11 at 8 54 39 AM](https://user-images.githubusercontent.com/3613392/76419011-063a5b80-6376-11ea-87a9-934e5300cc97.png)

### How To Test
1. Sign in to the app, swap to Hibana using the `appteam` account
1. Resize the window, verify the desktop nav renders at 960px and above, verify the mobile nav (just a placeholder at the moment) renders below 960px
1. Verify nav items render appropriately according to user permissions (toggling between the OG theme and the Hibana theme is a good way to check)
1. Verify that nav links with children nav items render those children in a sub navigation section with a dark background
1. Verify that the active state for links renders appropriately for each link in the app (in the case of subnav items, verify the primary nav
1. Verify the navigation is keyboard navigable - tab through the elements and use the enter key to navigate - also verify the skip navigation link moves the user to the main
1. Verify Heroku banner doesn't overlap the navigation. _Ask me for details on how to get this to render_

**NOTE:** Some navigation items are being moved elsewhere, and others exist in the navigation only temporarily:
- 'Alerts' and 'Configuration' are moving to the accounts dropdown (FE-924)
- Many subnav items will eventually be removed from the subnav as part of a re-architecture of the navigation (UX is not yet ready with that design, so it hasn't been implemented)
- Nav items currently have some unusual wrapping behavior due to the current navigation architecture - the plan is to resolve this issue as part of the updated inbox tracker rollout

### To Do
- [X] Double check alternate layouts
- [X] Fix loading positioning introduced by flexbox changes on the dashboard - **will be addressed as part of <Loading/> component update**
- [X] Check conditional routing to RV
- [X] Fix vertical alignment issues (go to `/dashboard` on a large-ish viewport)
- [x] Figure out how to offset `<BoomerangBanner/>`
- [ ] Does `<PendingCancelGlobalBanner />` need to be handled?
